### ISSUE TITLE
Yearly habitat validation 

### ISSUE THREAD

**Issue by RetoSchmucki** [BiologicalRecordsCentre/ABLE#519]
@JimBacon and @kazlauskis Currently, the French BMS system request the user to confirm or record any change in the habitat type along the monitoring sections. This feature requires habitat validation once every year before starting recording and is mandatory. Could this be implemented in the app and the online system?

**Comment by kazlauskis** [comment]
We don't collect habitat info in the current app version but it would be easy to add another field and make it mandatory. This would require users to do it for every section of all their surveys though - maybe a bit too much.

**Comment by CrisSevilleja** [comment]
I think it is a great idea and claimed by other schemes that don´t use habitat change at the moment. I would make filling this field of habitat mandatory the first time of the year by doing a transect and including it per section. For the rest of the visits by default no change in habitat but with the possibility to add pre-defined changes. 

What do you think @DavidRoy and @chrisvanswaay? there is something similar in the Dutch system where the recorder has to fill if there are changes per section on the habitat

**Comment by JimBacon** [comment]
The habitat information is part of the definition of the location, currently, and not part of the sample.
Once records have been made against a location it must not be changed otherwise it will make those existing records incorrect.

If a change of habitat needs to be recorded then a new location record is needed.

There is nothing to prevent this being done through the website currently. I can see there are features we could add to facilitate this, e.g. 'create copy of existing transect' 

I can also foresee this causing confusion because the transects will end up with names like 'Monks Wood', 'Monks Wood 2', 'Monks Wood 2023', 'Monks Wood 2025 to ?' We might need to add an attribute to indicate a location should no longer be used for data entry.

Of course we can change the way we record information and make habitat part of the sample if that is better.

**Comment by DavidRoy** [comment]
@JimBacon how about keeping the current approach but treating as 'habitat when location setup'. e.g. original habitat
Then any changes are captured through sample attributes?

**Comment by JimBacon** [comment]
Sure, @DavidRoy, we can have the habitat attribiutes on both the locations and the samples if you want.

If we add sample attributes for habitat (three for the transect and four for each section is what we have currently) then I can see it working in two ways.

1. The sample attributes are always populated which would be irksome to the recorder unless we make it default to the last values entered.
2. The sample attributes are populated sparsely so that, to determine the habitat for a visit, you have to look back and find the most recent previous visit where habitat was recorded.

The first sounds like less trouble to maintain.

There are other attributes which are also stored against the location currently which must be just as likely to change over time as the habitat, namely land management and land tenure. You should be considering whether these should be sample attributes too.

**Comment by CrisSevilleja** [comment]
Thanks for the ideas on how to do it @JimBacon but I want to be sure what is exactly we are talking about. When @RetoSchmucki mentioned "record any change in the habitat" are we talking about confirming the EUNIS habitat for example? or are we talking about land management, changes in the habitat like mowing, fire or abandonment? 

I am interested in both, and I want to know what is going to be added for the volunteers. At the moment, we register the habitat of the transect but in a weak way with a super general standardization but it would be better to include EUNIS or the UICN classification. In that case, we cannot make this sample attribute compulsory, due to it may create confusion for the general users.

Another thing I was referring to more is the changes in the habitat, quite interested in this, but first is has to be set up properly the habitat of the transect.

**Comment by DavidRoy** [comment]
@JimBacon let's discuss next week.  The habitat and management will not change very often so will not be mandatory.  I suggest the following workflow:
1. For first visit each year, recorders are asked to confirm habitat and management information for transect.  There is a data entry form for confirming the information.  Need to decide if we store new sample attribute information if nothing has changed.  Ideally not.
2. On the visit data entry page there is a tick box for 'Update habitat and management information'.  If selected then a data entry form is provided to update the information, populated with the last values entered (could be either from the location attributes or sample attributes)
